bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AccessibleToolItemBuilder.java

```diff
@@ -0,0 +1,61 @@
+package org.eclipse.ui.texteditor;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.widgets.ToolItem;
+
+/**
+ * Builder for ToolItems for {@link AccessibleToolBars}.
+ *
+ * @since 3.17
+ */
+public class AccessibleToolItemBuilder {
+	private AccessibleToolBar accessibleToolBar;
+	private int styleBits = SWT.NONE;
+	private Image image;
+	private String toolTipText;
+	private SelectionListener selectionListener;
+
+	public AccessibleToolItemBuilder(AccessibleToolBar accessibleToolBar) {
+	        this.accessibleToolBar = accessibleToolBar;
```

Comment:
```suggestion
	        this.accessibleToolBar = Objects.requireNonNull(accessibleToolBar);
```

Reply:
I did not know you could do this! Thanks :)

Reply:
Done in https://github.com/eclipse-platform/eclipse.platform.ui/commit/0c0d22bc8f0ea13eb6e1124c89747f7a4f02e266